internal/xds/translator/listener.go

```diff
@@ -930,18 +930,26 @@ func buildALPNProtocols(alpn []string) []string {
 	}
 }
 
-func buildXdsTLSCertSecret(tlsConfig ir.TLSCertificate) *tlsv3.Secret {
+func buildXdsTLSCertSecret(tlsConfig *ir.TLSCertificate) *tlsv3.Secret {
```

Comment:
hugeParam: tlsConfig is heavy (88 bytes); consider passing it by pointer (gocritic)